From 5126eda91465965fb17815554df07986bd7b3ed5 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 9 Dec 2024 14:08:22 +0100
Subject: [PATCH 1/4] Add Folia Supported plugin.yml

---
 Insights/build.gradle.kts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Insights/build.gradle.kts b/Insights/build.gradle.kts
index a4387ce..bfe41c8 100644
--- a/Insights/build.gradle.kts
+++ b/Insights/build.gradle.kts
@@ -36,6 +36,7 @@ bukkit {
     website = "https://github.com/InsightsPlugin/Insights"
     softDepend = listOf("PlaceholderAPI")
     authors = listOf("FrankHeijden")
+    foliaSupported = true
     permissions {
         register("insights.info") {
             description = "Allows you to see information about insights"
-- 
2.47.1.windows.1

