From e1904538c65507078e0f5fdae305b06207653ec4 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Sun, 26 Jan 2025 13:05:43 +0100
Subject: [PATCH 2/6] Add Folia Supported plugin.yml

---
 Insights/build.gradle.kts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Insights/build.gradle.kts b/Insights/build.gradle.kts
index a4387ce..bfe41c8 100644
--- a/Insights/build.gradle.kts
+++ b/Insights/build.gradle.kts
@@ -36,6 +36,7 @@ bukkit {
     website = "https://github.com/InsightsPlugin/Insights"
     softDepend = listOf("PlaceholderAPI")
     authors = listOf("FrankHeijden")
+    foliaSupported = true
     permissions {
         register("insights.info") {
             description = "Allows you to see information about insights"
-- 
2.47.1.windows.2

