From f52f182ae770ea1ab44107bb8c0600b8297bd879 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 16 Jun 2025 03:32:44 +0200
Subject: [PATCH 2/9] Add Folia Supported plugin.yml

---
 Insights/build.gradle.kts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Insights/build.gradle.kts b/Insights/build.gradle.kts
index a4387ce..bfe41c8 100644
--- a/Insights/build.gradle.kts
+++ b/Insights/build.gradle.kts
@@ -36,6 +36,7 @@ bukkit {
     website = "https://github.com/InsightsPlugin/Insights"
     softDepend = listOf("PlaceholderAPI")
     authors = listOf("FrankHeijden")
+    foliaSupported = true
     permissions {
         register("insights.info") {
             description = "Allows you to see information about insights"
-- 
2.49.0.windows.1

