From 7c4a181cbeffc8922b7edca861d6ab4a1005e19a Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Mon, 27 Apr 2026 19:26:20 +0200
Subject: [PATCH 03/14] Patch ScanTask

---
 .../insights/api/tasks/ScanTask.java          | 69 ++++++++++---------
 1 file changed, 38 insertions(+), 31 deletions(-)

diff --git a/Insights-API/src/main/java/dev/frankheijden/insights/api/tasks/ScanTask.java b/Insights-API/src/main/java/dev/frankheijden/insights/api/tasks/ScanTask.java
index fa76919..606649a 100644
--- a/Insights-API/src/main/java/dev/frankheijden/insights/api/tasks/ScanTask.java
+++ b/Insights-API/src/main/java/dev/frankheijden/insights/api/tasks/ScanTask.java
@@ -14,6 +14,7 @@ import dev.frankheijden.insights.api.util.TriConsumer;
 import dev.frankheijden.insights.api.utils.EnumUtils;
 import dev.frankheijden.insights.api.utils.StringUtils;
 import net.kyori.adventure.text.Component;
+import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
 import org.bukkit.scheduler.BukkitScheduler;
 import org.bukkit.scheduler.BukkitTask;
@@ -26,6 +27,7 @@ import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Consumer;
@@ -51,7 +53,7 @@ public class ScanTask<R> implements Runnable {
     private final AtomicBoolean completedExceptionally = new AtomicBoolean();
     private final int chunkCount;
     private long lastInfo = 0;
-    private BukkitTask task;
+    private io.papermc.paper.threadedregions.scheduler.ScheduledTask task;
 
     /**
      * Creates a new ScanTask to scan a collection of ChunkPart's.
@@ -370,8 +372,8 @@ public class ScanTask<R> implements Runnable {
     }
 
     private void start() {
-        BukkitScheduler scheduler = plugin.getServer().getScheduler();
-        task = scheduler.runTaskTimer(plugin, this, 0, plugin.getSettings().SCANS_ITERATION_INTERVAL_TICKS);
+        //BukkitScheduler scheduler = plugin.getServer().getScheduler();
+        task = plugin.getServer().getAsyncScheduler().runAtFixedRate(plugin, scheduledTask -> this.run(), 50L, plugin.getSettings().SCANS_ITERATION_INTERVAL_TICKS * 50L, TimeUnit.MILLISECONDS);
     }
 
     private void cancel() {
@@ -406,6 +408,8 @@ public class ScanTask<R> implements Runnable {
         iterationChunks.addAndGet(-chunkIterations);
 
         // Iterate 'chunkIterations' times
+        int delayIncrement = 1;
+        int batchSize = 16;
         for (var i = 0; i < chunkIterations; i++) {
             // Note: we can't cancel the task here just yet,
             // because some chunks might still need scanning (after loading).
@@ -416,35 +420,38 @@ public class ScanTask<R> implements Runnable {
             var loc = chunkPart.getChunkLocation();
             var world = loc.getWorld();
 
-            CompletableFuture<Storage> storageFuture;
-            if (world.isChunkLoaded(loc.getX(), loc.getZ())) {
-                storageFuture = executor.submit(
-                        world.getChunkAt(loc.getX(), loc.getZ()),
-                        chunkPart.getChunkCuboid(),
-                        options
-                );
-            } else {
-                storageFuture = executor.submit(
-                        loc.getWorld(),
-                        loc.getX(),
-                        loc.getZ(),
-                        chunkPart.getChunkCuboid(),
-                        options
-                );
-            }
+            long delay = (long) ((i / (double) batchSize) * delayIncrement);
+            Bukkit.getRegionScheduler().runDelayed(plugin, world, loc.getX(), loc.getZ(), scheduledTask -> {
+                CompletableFuture<Storage> storageFuture;
+                if (world.isChunkLoaded(loc.getX(), loc.getZ())) {
+                    storageFuture = executor.submit(
+                            world.getChunkAt(loc.getX(), loc.getZ()),
+                            chunkPart.getChunkCuboid(),
+                            options
+                    );
+                } else {
+                    storageFuture = executor.submit(
+                            loc.getWorld(),
+                            loc.getX(),
+                            loc.getZ(),
+                            chunkPart.getChunkCuboid(),
+                            options
+                    );
+                }
 
-            storageFuture
-                    .thenAccept(storage -> resultMerger.accept(storage, loc, result))
-                    .thenRun(() -> {
-                        iterationChunks.incrementAndGet();
-                        chunks.incrementAndGet();
-                    })
-                    .exceptionally(th -> {
-                        if (!completedExceptionally.getAndSet(true)) {
-                            plugin.getLogger().log(Level.SEVERE, th, th::getMessage);
-                        }
-                        return null;
-                    });
+                storageFuture
+                        .thenAccept(storage -> resultMerger.accept(storage, loc, result))
+                        .thenRun(() -> {
+                            iterationChunks.incrementAndGet();
+                            chunks.incrementAndGet();
+                        })
+                        .exceptionally(th -> {
+                            if (!completedExceptionally.getAndSet(true)) {
+                                plugin.getLogger().log(Level.SEVERE, th, th::getMessage);
+                            }
+                            return null;
+                        });
+            }, Math.max(1, delay));
         }
     }
 
-- 
2.52.0.windows.1

