From 9635d66988a8fbbfbe3da6682239afc71592321d Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 9 Dec 2024 14:08:23 +0100
Subject: [PATCH 4/4] Fix piston push

---
 .../insights/listeners/BlockListener.java           | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/Insights/src/main/java/dev/frankheijden/insights/listeners/BlockListener.java b/Insights/src/main/java/dev/frankheijden/insights/listeners/BlockListener.java
index 4ecf365..6cd374d 100644
--- a/Insights/src/main/java/dev/frankheijden/insights/listeners/BlockListener.java
+++ b/Insights/src/main/java/dev/frankheijden/insights/listeners/BlockListener.java
@@ -340,11 +340,14 @@ public class BlockListener extends InsightsListener {
 
         plugin.getServer().getGlobalRegionScheduler().runDelayed(plugin, (scheduledTask) -> {
             for (var i = 0; i < blocks.size(); i++) {
-                var relative = blocks.get(i).getRelative(event.getDirection());
-                var material = relative.getType();
-                if (materials[i] == material) {
-                    handleModification(relative.getLocation(), material, 1);
-                }
+                final var iteration = i;
+                var relative = blocks.get(iteration).getRelative(event.getDirection());
+                plugin.getServer().getRegionScheduler().execute(plugin, relative.getLocation(), () -> {
+                    var material = relative.getType();
+                    if (materials[iteration] == material) {
+                        handleModification(relative.getLocation(), material, 1);
+                    }
+                });
             }
         }, 3L); // Ensures blocks have been updated after piston move
     }
-- 
2.47.1.windows.1

