From 2eeb5e400f2f058585dab8a320fbf9bc03e0ebed Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 16 Jun 2025 03:24:28 +0200
Subject: [PATCH 7/9] Fix scheduler scan task

---
 .../insights/api/tasks/ScanTask.java          | 47 ++++++++-----------
 1 file changed, 19 insertions(+), 28 deletions(-)

diff --git a/Insights-API/src/main/java/dev/frankheijden/insights/api/tasks/ScanTask.java b/Insights-API/src/main/java/dev/frankheijden/insights/api/tasks/ScanTask.java
index 07ce1aa..5e31609 100644
--- a/Insights-API/src/main/java/dev/frankheijden/insights/api/tasks/ScanTask.java
+++ b/Insights-API/src/main/java/dev/frankheijden/insights/api/tasks/ScanTask.java
@@ -418,35 +418,26 @@ public class ScanTask<R> implements Runnable {
             var loc = chunkPart.getChunkLocation();
             var world = loc.getWorld();
 
-            CompletableFuture<Storage> storageFuture;
-            if (world.isChunkLoaded(loc.getX(), loc.getZ())) {
-                storageFuture = executor.submit(
-                        world.getChunkAt(loc.getX(), loc.getZ()),
-                        chunkPart.getChunkCuboid(),
-                        options
-                );
-            } else {
-                storageFuture = executor.submit(
-                        loc.getWorld(),
-                        loc.getX(),
-                        loc.getZ(),
-                        chunkPart.getChunkCuboid(),
-                        options
-                );
-            }
+            org.bukkit.Bukkit.getRegionScheduler().execute(this.plugin, world, loc.getX(), loc.getZ(), () -> {
+                CompletableFuture<Storage> storageFuture;
+                if (world.isChunkLoaded(loc.getX(), loc.getZ())) {
+                    storageFuture = executor
+                            .submit(world.getChunkAt(loc.getX(), loc.getZ()), chunkPart.getChunkCuboid(), this.options);
+                } else {
+                    storageFuture = executor
+                            .submit(loc.getWorld(), loc.getX(), loc.getZ(), chunkPart.getChunkCuboid(), this.options);
+                }
 
-            storageFuture
-                    .thenAccept(storage -> resultMerger.accept(storage, loc, result))
-                    .thenRun(() -> {
-                        iterationChunks.incrementAndGet();
-                        chunks.incrementAndGet();
-                    })
-                    .exceptionally(th -> {
-                        if (!completedExceptionally.getAndSet(true)) {
-                            plugin.getLogger().log(Level.SEVERE, th, th::getMessage);
-                        }
-                        return null;
-                    });
+                storageFuture.thenAccept((var) -> this.resultMerger.accept(var, loc, this.result)).thenRun(() -> {
+                    this.iterationChunks.incrementAndGet();
+                    this.chunks.incrementAndGet();
+                }).exceptionally((th) -> {
+                    if (!this.completedExceptionally.getAndSet(true)) {
+                        this.plugin.getLogger().log(Level.SEVERE, th, th::getMessage);
+                    }
+                    return null;
+                });
+            });
         }
     }
 
-- 
2.49.0.windows.1

