From 03d4aaf1d92995030150757ccb340ec6f11b7181 Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Mon, 27 Apr 2026 19:26:21 +0200
Subject: [PATCH 07/14] Patch CommandScan

---
 .../insights/commands/CommandScan.java        | 23 +++++++++++--------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/Insights-Core/src/main/java/dev/frankheijden/insights/commands/CommandScan.java b/Insights-Core/src/main/java/dev/frankheijden/insights/commands/CommandScan.java
index ea1ad92..c74d000 100644
--- a/Insights-Core/src/main/java/dev/frankheijden/insights/commands/CommandScan.java
+++ b/Insights-Core/src/main/java/dev/frankheijden/insights/commands/CommandScan.java
@@ -9,6 +9,7 @@ import dev.frankheijden.insights.api.reflection.RTileEntityTypes;
 import dev.frankheijden.insights.api.tasks.ScanTask;
 import dev.frankheijden.insights.api.util.LazyChunkPartRadiusIterator;
 import dev.frankheijden.insights.api.utils.Constants;
+import org.bukkit.Bukkit;
 import org.bukkit.Chunk;
 import org.bukkit.World;
 import org.bukkit.entity.Player;
@@ -114,17 +115,19 @@ public class CommandScan extends InsightsCommand {
             boolean displayZeros,
             boolean groupByChunk
     ) {
-        Chunk chunk = player.getLocation().getChunk();
-        World world = chunk.getWorld();
-        int chunkX = chunk.getX();
-        int chunkZ = chunk.getZ();
+        Bukkit.getRegionScheduler().run(plugin, player.getLocation(), scheduledTask -> {
+            Chunk chunk = player.getLocation().getChunk();
+            World world = chunk.getWorld();
+            int chunkX = chunk.getX();
+            int chunkZ = chunk.getZ();
 
-        LazyChunkPartRadiusIterator it = new LazyChunkPartRadiusIterator(world, chunkX, chunkZ, radius);
+            LazyChunkPartRadiusIterator it = new LazyChunkPartRadiusIterator(world, chunkX, chunkZ, radius);
 
-        if (groupByChunk) {
-            ScanTask.scanAndDisplayGroupedByChunk(plugin, player, it, it.getChunkCount(), options, items, false);
-        } else {
-            ScanTask.scanAndDisplay(plugin, player, it, it.getChunkCount(), options, items, displayZeros);
-        }
+            if (groupByChunk) {
+                ScanTask.scanAndDisplayGroupedByChunk(plugin, player, it, it.getChunkCount(), options, items, false);
+            } else {
+                ScanTask.scanAndDisplay(plugin, player, it, it.getChunkCount(), options, items, displayZeros);
+            }
+        });
     }
 }
-- 
2.52.0.windows.1

