From b55c7d3da8acd5f4c30336e25c160b26bcf52044 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Wed, 17 Dec 2025 18:55:41 +0100
Subject: [PATCH 09/13] Patch PlayerTrackerTask

---
 .../insights/tasks/PlayerTrackerTask.java     | 34 ++++++++++---------
 1 file changed, 18 insertions(+), 16 deletions(-)

diff --git a/Insights-Core/src/main/java/dev/frankheijden/insights/tasks/PlayerTrackerTask.java b/Insights-Core/src/main/java/dev/frankheijden/insights/tasks/PlayerTrackerTask.java
index 9603f13..d26079a 100644
--- a/Insights-Core/src/main/java/dev/frankheijden/insights/tasks/PlayerTrackerTask.java
+++ b/Insights-Core/src/main/java/dev/frankheijden/insights/tasks/PlayerTrackerTask.java
@@ -46,29 +46,31 @@ public class PlayerTrackerTask extends InsightsAsyncTask {
             return;
         }
 
-        plugin.getServer().getScheduler().runTask(plugin, () -> {
+        plugin.getServer().getGlobalRegionScheduler().execute(plugin, () -> {
             long now = System.nanoTime();
             for (ChunkLocation loc : locations) {
                 var world = loc.getWorld();
                 if (world.isChunkLoaded(loc.getX(), loc.getZ())) {
                     this.scanLocations.put(loc, now);
 
-                    var chunk = world.getChunkAt(loc.getX(), loc.getZ());
-                    plugin.getChunkContainerExecutor().submit(chunk, ScanOptions.all()).whenComplete((s, e) -> {
-                        if (s == null) {
-                            int hash = e.getStackTrace()[0].hashCode();
-                            if (!knownErrorStackTraceHashes.contains(hash)) {
-                                knownErrorStackTraceHashes.add(hash);
-                                plugin.getLogger().log(
-                                        Level.SEVERE,
-                                        "Error occurred while scanning "
-                                                + loc
-                                                + " (future errors with the same stacktrace are suppressed)",
-                                        e
-                                );
+                    plugin.getServer().getRegionScheduler().run(plugin, loc.getWorld(), loc.getX(), loc.getZ(), scheduledTask -> {
+                        var chunk = world.getChunkAt(loc.getX(), loc.getZ());
+                        plugin.getChunkContainerExecutor().submit(chunk, ScanOptions.all()).whenComplete((s, e) -> {
+                            if (s == null) {
+                                int hash = e.getStackTrace()[0].hashCode();
+                                if (!knownErrorStackTraceHashes.contains(hash)) {
+                                    knownErrorStackTraceHashes.add(hash);
+                                    plugin.getLogger().log(
+                                            Level.SEVERE,
+                                            "Error occurred while scanning "
+                                                    + loc
+                                                    + " (future errors with the same stacktrace are suppressed)",
+                                            e
+                                    );
+                                }
                             }
-                        }
-                        this.scanLocations.remove(loc);
+                            this.scanLocations.remove(loc);
+                        });
                     });
                 }
             }
-- 
2.52.0.windows.1

