From 1563bdfc3e6e233e36c2c627676d78d03d1aad2b Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Sat, 25 Oct 2025 20:12:01 +0200
Subject: [PATCH 11/21] Patch Insights src main java dev frankheijden insigh

---
 .../insights/commands/CommandScan.java        | 22 ++++++++++---------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/Insights/src/main/java/dev/frankheijden/insights/commands/CommandScan.java b/Insights/src/main/java/dev/frankheijden/insights/commands/CommandScan.java
index ea1ad92..9c0cfec 100644
--- a/Insights/src/main/java/dev/frankheijden/insights/commands/CommandScan.java
+++ b/Insights/src/main/java/dev/frankheijden/insights/commands/CommandScan.java
@@ -114,17 +114,19 @@ public class CommandScan extends InsightsCommand {
             boolean displayZeros,
             boolean groupByChunk
     ) {
-        Chunk chunk = player.getLocation().getChunk();
-        World world = chunk.getWorld();
-        int chunkX = chunk.getX();
-        int chunkZ = chunk.getZ();
+        org.bukkit.Bukkit.getRegionScheduler().execute(plugin, player.getLocation(), () -> {
+            Chunk chunk = player.getLocation().getChunk();
+            World world = chunk.getWorld();
+            int chunkX = chunk.getX();
+            int chunkZ = chunk.getZ();
 
-        LazyChunkPartRadiusIterator it = new LazyChunkPartRadiusIterator(world, chunkX, chunkZ, radius);
+            LazyChunkPartRadiusIterator it = new LazyChunkPartRadiusIterator(world, chunkX, chunkZ, radius);
 
-        if (groupByChunk) {
-            ScanTask.scanAndDisplayGroupedByChunk(plugin, player, it, it.getChunkCount(), options, items, false);
-        } else {
-            ScanTask.scanAndDisplay(plugin, player, it, it.getChunkCount(), options, items, displayZeros);
-        }
+            if (groupByChunk) {
+                ScanTask.scanAndDisplayGroupedByChunk(plugin, player, it, it.getChunkCount(), options, items, false);
+            } else {
+                ScanTask.scanAndDisplay(plugin, player, it, it.getChunkCount(), options, items, displayZeros);
+            }
+        });
     }
 }
-- 
2.50.1.windows.1

