From fde05b68056555b2373eb877ec99b95043d52412 Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Mon, 27 Apr 2026 19:26:22 +0200
Subject: [PATCH 11/14] Patch TickResetCount

---
 .../api/concurrent/count/TickResetCount.java         | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/Insights-API/src/main/java/dev/frankheijden/insights/api/concurrent/count/TickResetCount.java b/Insights-API/src/main/java/dev/frankheijden/insights/api/concurrent/count/TickResetCount.java
index 7a7dc27..d3a3ad6 100644
--- a/Insights-API/src/main/java/dev/frankheijden/insights/api/concurrent/count/TickResetCount.java
+++ b/Insights-API/src/main/java/dev/frankheijden/insights/api/concurrent/count/TickResetCount.java
@@ -4,6 +4,7 @@ import dev.frankheijden.insights.api.InsightsPlugin;
 import org.bukkit.scheduler.BukkitTask;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
 
 public class TickResetCount<T> {
 
@@ -12,7 +13,7 @@ public class TickResetCount<T> {
     private final int size;
     private final Map<T, IntegerCount> counts;
     private final ResetTask resetTask;
-    private BukkitTask bukkitTask = null;
+    private io.papermc.paper.threadedregions.scheduler.ScheduledTask bukkitTask = null;
 
     /**
      * Constructs a new TickResetCount.
@@ -34,11 +35,12 @@ public class TickResetCount<T> {
             throw new IllegalStateException("ResetTask is already running!");
         }
 
-        this.bukkitTask = plugin.getServer().getScheduler().runTaskTimerAsynchronously(
+        this.bukkitTask = plugin.getServer().getAsyncScheduler().runAtFixedRate(
                 plugin,
-                resetTask,
-                0,
-                intervalTicks
+                scheduledTask -> resetTask.run(),
+                50L,
+                intervalTicks *50L,
+                TimeUnit.MILLISECONDS
         );
     }
 
-- 
2.52.0.windows.1

