From 1ea588c5d7d5e37026bfd1e6742fee0072300eca Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Wed, 17 Dec 2025 19:01:22 +0100
Subject: [PATCH 12/13] Patch Insights

---
 .../dev/frankheijden/insights/Insights.java   | 25 +++++++++++--------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/Insights-Core/src/main/java/dev/frankheijden/insights/Insights.java b/Insights-Core/src/main/java/dev/frankheijden/insights/Insights.java
index cfe67e7..cbf69fb 100644
--- a/Insights-Core/src/main/java/dev/frankheijden/insights/Insights.java
+++ b/Insights-Core/src/main/java/dev/frankheijden/insights/Insights.java
@@ -58,6 +58,7 @@ import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.Locale;
+import java.util.concurrent.TimeUnit;
 
 public class Insights extends InsightsPlugin {
 
@@ -83,8 +84,8 @@ public class Insights extends InsightsPlugin {
     private ScanHistory scanHistory;
     private ListenerManager listenerManager;
     private InsightsPlaceholderExpansion placeholderExpansion;
-    private BukkitTask playerTracker = null;
-    private BukkitTask updateChecker = null;
+    private io.papermc.paper.threadedregions.scheduler.ScheduledTask playerTracker = null;
+    private io.papermc.paper.threadedregions.scheduler.ScheduledTask updateChecker = null;
     private BukkitAudiences audiences = null;
     private RedstoneUpdateCount redstoneUpdateCount = null;
     private ChunkTeleport chunkTeleport;
@@ -117,7 +118,7 @@ public class Insights extends InsightsPlugin {
             ex.printStackTrace();
         }
 
-        getServer().getScheduler().runTaskLater(this, () -> {
+        getServer().getGlobalRegionScheduler().runDelayed(this, (scheduledTask) -> {
             try {
                 addonManager.loadAddons();
             } catch (IOException ex) {
@@ -379,11 +380,12 @@ public class Insights extends InsightsPlugin {
         }
 
         if (settings.CHUNK_SCANS_MODE == Settings.ChunkScanMode.ALWAYS) {
-            playerTracker = getServer().getScheduler().runTaskTimerAsynchronously(
+            playerTracker = getServer().getAsyncScheduler().runAtFixedRate(
                     this,
-                    new PlayerTrackerTask(this),
-                    0,
-                    settings.CHUNK_SCANS_PLAYER_TRACKER_INTERVAL_TICKS
+                    scheduledTask -> new PlayerTrackerTask(this).run(),
+                    50L,
+                    settings.CHUNK_SCANS_PLAYER_TRACKER_INTERVAL_TICKS * 50L,
+                    TimeUnit.MILLISECONDS
             );
         }
 
@@ -392,11 +394,12 @@ public class Insights extends InsightsPlugin {
         }
 
         if (settings.UPDATE_CHECKER_ENABLED) {
-            updateChecker = getServer().getScheduler().runTaskTimerAsynchronously(
+            updateChecker = getServer().getAsyncScheduler().runAtFixedRate(
                     this,
-                    new UpdateCheckerTask(this),
-                    20,
-                    20L * settings.UPDATE_CHECKER_INTERVAL_SECONDS
+                    scheduledTask -> new UpdateCheckerTask(this).run(),
+                    20 * 50L,
+                    20L * settings.UPDATE_CHECKER_INTERVAL_SECONDS * 50L,
+                    TimeUnit.MILLISECONDS
             );
         }
 
-- 
2.52.0.windows.1

