From 9cf671ea0e83e4864d89724a66396a09d8cdde4f Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Thu, 4 Sep 2025 00:03:45 +0200
Subject: [PATCH 13/20] Patch Insights src main java dev frankheijden insigh

---
 .../insights/tasks/PlayerTrackerTask.java     | 35 ++++++++++---------
 1 file changed, 18 insertions(+), 17 deletions(-)

diff --git a/Insights/src/main/java/dev/frankheijden/insights/tasks/PlayerTrackerTask.java b/Insights/src/main/java/dev/frankheijden/insights/tasks/PlayerTrackerTask.java
index 9603f13..9e73126 100644
--- a/Insights/src/main/java/dev/frankheijden/insights/tasks/PlayerTrackerTask.java
+++ b/Insights/src/main/java/dev/frankheijden/insights/tasks/PlayerTrackerTask.java
@@ -46,29 +46,30 @@ public class PlayerTrackerTask extends InsightsAsyncTask {
             return;
         }
 
-        plugin.getServer().getScheduler().runTask(plugin, () -> {
+        plugin.getServer().getGlobalRegionScheduler().execute(plugin, () -> {
             long now = System.nanoTime();
             for (ChunkLocation loc : locations) {
                 var world = loc.getWorld();
                 if (world.isChunkLoaded(loc.getX(), loc.getZ())) {
                     this.scanLocations.put(loc, now);
-
-                    var chunk = world.getChunkAt(loc.getX(), loc.getZ());
-                    plugin.getChunkContainerExecutor().submit(chunk, ScanOptions.all()).whenComplete((s, e) -> {
-                        if (s == null) {
-                            int hash = e.getStackTrace()[0].hashCode();
-                            if (!knownErrorStackTraceHashes.contains(hash)) {
-                                knownErrorStackTraceHashes.add(hash);
-                                plugin.getLogger().log(
-                                        Level.SEVERE,
-                                        "Error occurred while scanning "
-                                                + loc
-                                                + " (future errors with the same stacktrace are suppressed)",
-                                        e
-                                );
+                    org.bukkit.Bukkit.getRegionScheduler().execute(plugin, world, loc.getX(), loc.getZ(), () -> {
+                        var chunk = world.getChunkAt(loc.getX(), loc.getZ());
+                        plugin.getChunkContainerExecutor().submit(chunk, ScanOptions.all()).whenComplete((s, e) -> {
+                            if (s == null) {
+                                int hash = e.getStackTrace()[0].hashCode();
+                                if (!knownErrorStackTraceHashes.contains(hash)) {
+                                    knownErrorStackTraceHashes.add(hash);
+                                    plugin.getLogger().log(
+                                            Level.SEVERE,
+                                            "Error occurred while scanning "
+                                                    + loc
+                                                    + " (future errors with the same stacktrace are suppressed)",
+                                            e
+                                    );
+                                }
                             }
-                        }
-                        this.scanLocations.remove(loc);
+                            this.scanLocations.remove(loc);
+                        });
                     });
                 }
             }
-- 
2.50.1.windows.1

