From 39b8c60063eb0ebbbfc806c54314ee33bd7b864f Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Mon, 27 Apr 2026 19:26:22 +0200
Subject: [PATCH 13/14] Patch Core build.gradle.kts

---
 Insights-Core/build.gradle.kts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Insights-Core/build.gradle.kts b/Insights-Core/build.gradle.kts
index c1c5f0b..ea0f3de 100644
--- a/Insights-Core/build.gradle.kts
+++ b/Insights-Core/build.gradle.kts
@@ -37,6 +37,7 @@ bukkit {
     website = "https://github.com/InsightsPlugin/Insights"
     softDepend = listOf("PlaceholderAPI")
     authors = listOf("FrankHeijden", "Jsinco")
+    foliaSupported = true
     permissions {
         register("insights.info") {
             description = "Allows you to see information about insights"
-- 
2.52.0.windows.1

