From a4b0eaf0658488958c9912c2e303e5481149f0c2 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Thu, 4 Sep 2025 00:03:47 +0200
Subject: [PATCH 17/20] Patch Insights build.gradle.kts

---
 Insights/build.gradle.kts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Insights/build.gradle.kts b/Insights/build.gradle.kts
index a4387ce..30f163b 100644
--- a/Insights/build.gradle.kts
+++ b/Insights/build.gradle.kts
@@ -35,6 +35,7 @@ bukkit {
     apiVersion = "1.21"
     website = "https://github.com/InsightsPlugin/Insights"
     softDepend = listOf("PlaceholderAPI")
+    foliaSupported = true
     authors = listOf("FrankHeijden")
     permissions {
         register("insights.info") {
-- 
2.50.1.windows.1

