From 167a47c85338a632cd9011c8672782dce5c8bdc7 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Sat, 25 Oct 2025 20:18:47 +0200
Subject: [PATCH 21/21] Fix InsightsNMSImpl

---
 .../insights/nms/impl/InsightsNMSImpl.java             | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/Insights-NMS/Current/src/main/java/dev/frankheijden/insights/nms/impl/InsightsNMSImpl.java b/Insights-NMS/Current/src/main/java/dev/frankheijden/insights/nms/impl/InsightsNMSImpl.java
index bff9322..219d7df 100644
--- a/Insights-NMS/Current/src/main/java/dev/frankheijden/insights/nms/impl/InsightsNMSImpl.java
+++ b/Insights-NMS/Current/src/main/java/dev/frankheijden/insights/nms/impl/InsightsNMSImpl.java
@@ -88,8 +88,9 @@ public class InsightsNMSImpl extends InsightsNMS {
             if (sectionIndex < 0 || sectionIndex >= sectionsCount) continue;
 
             PalettedContainer<BlockState> blockStateContainer;
+            net.minecraft.world.level.chunk.Strategy<BlockState> strategy = serverLevel.palettedContainerFactory().blockStatesStrategy();
             if (sectionTag.contains("block_states")) {
-                Codec<PalettedContainer<BlockState>> blockStateCodec = SerializableChunkData.BLOCK_STATE_CODEC;
+                Codec<PalettedContainer<BlockState>> blockStateCodec = PalettedContainer.codecRW(BlockState.CODEC, strategy, Blocks.AIR.defaultBlockState(), new BlockState[0]);
                 dataResult = blockStateCodec.parse(NbtOps.INSTANCE, sectionTag.getCompound("block_states").orElseThrow())
                         .promotePartial(message -> logger.severe(String.format(
                         CHUNK_ERROR,
@@ -106,12 +107,7 @@ public class InsightsNMSImpl extends InsightsNMS {
                     throw ex;
                 }
             } else {
-                blockStateContainer = new PalettedContainer<>(
-                        Block.BLOCK_STATE_REGISTRY,
-                        Blocks.AIR.defaultBlockState(),
-                        PalettedContainer.Strategy.SECTION_STATES,
-                        null
-                );
+                blockStateContainer = new PalettedContainer<>(Blocks.AIR.defaultBlockState(), strategy, new BlockState[0]);
             }
 
             LevelChunkSection chunkSection = new LevelChunkSection(blockStateContainer, null);
-- 
2.50.1.windows.1

